pymc3/distributions/multivariate.py

```diff
@@ -486,23 +486,9 @@ class Dirichlet(Continuous):
 
     def __init__(self, a, transform=transforms.stick_breaking,
                  *args, **kwargs):
-
-        if not isinstance(a, pm.model.TensorVariable):
-            if not isinstance(a, list) and not isinstance(a, np.ndarray):
-                raise TypeError(
-                    'The vector of concentration parameters (a) must be a python list '
-                    'or numpy array.')
-            a = np.array(a)
-            if (a <= 0).any():
-               raise ValueError("All concentration parameters (a) must be > 0.")
```

Comment:
This means we don't check for strict positivity of `a`, doesn't it? If so, I think we should try to retain this behavior, at best for every type of `a`, or at least when the concentration parameter isn't a symbolic variable -- which is why I tried to do, but it seems like `pm.model.TensorVariable` doesn't cast as wide a net as I thought.

Reply:
We technically *can't* perform a consistent version of this check.  Even when we can find/extract NumPy arrays for a symbolic `a`, this wouldn't be consistent, since&mdash;for instance&mdash;we have shared variables that be changed after this check occurs.

Again, such a check is antithetical to use of a symbolic backend.

Reply:
Yeah, I clearly understand the difficulty of doing that with a symbolic backend, thanks 👌 My point was more about cases when `a` wasn't a symbolic variable, but it would mean more Theano circumvention logic -- it's true it's a choice.

Reply:
If we could enable this functionality at a more general level for all distributions (e.g. in `Distribution.__init__`, when a shape can be determined non-symbolically and it isn't given explicitly, do it), then I would likely be onboard, but to do it in only _some_ cases doesn't sound good.

Also, such a change should definitely be in its own PR.

Reply:
That would indeed be very nice!